### DISCUSSION TITLE
Ignore Album-Specific Tags for a Designated Folder of Singles

### DISCUSSION THREAD

**Discussion by michaelkrieger** [navidrome/navidrome#2853] [Ideas]
Possibly something that would be amazing in #2709 , combined with #1394 , is allow us to mark a folder as singles.  I have folders of Albums (Albums/A/Artist/Artist - Album (2009)/01 Title.mp3), and a few folders of “singles” (Singles/2009/Artist - Song.mp3). I would love to strip the album and album artist info from these files without losing the metadata from the file.  Knowing that the track came from an album I don’t have is good info, but I don’t want all of these single-song-from or two-songs-from albums cluttering my albums list. I’d be fine bringing in this info, but just not showing it in the list of albums. 

Retagging makes sense but I do like having the album information. I just don’t want lots of partial albums listed.

**Comment by dweymouth** [comment]
One complication with this is that the Subsonic API - which 3rd party clients use to interact with Navidrome - has no concept of songs that are not attached to an album. So they need to be assigned _some_ album to even be available at all to clients. Maybe Navidrome could create a special "Singles" album with "Various Artists" as the album artist, that contains all the albumless singles in the library, similar to how Various Artists is already handled as a special case for compilation albums?

**Reply by michaelkrieger** [reply]
I'd be very good with attaching them to a "( Singleton )" album with "Various Artists" and squashing the year if necessary.  It would just bridge the gap of a folder of singles not creating huge amount of albums.

**Comment by michaelkrieger** [comment]
I also thing this would be best created through an .ndignore file or maybe .ndsingles.  All we would really have is:
```
If the file is below a folder containing .ndsingles
    set the album artist = 'Various Artists'
    set album name = '(( Singleton ))' or '(( ParentFoldername ))'
    set year = ''
```
when reading the ID3 tags.  Would just be an override of the metadata.

**Comment by michaelkrieger** [comment]
Adding a further to this for more use cases.

I've recently pulled the files from my playlists from a popular music service.  As a result, I now have a bunch of folder sets like this: `/downloaded playlist files/my awesome playlist/123 - Artist - Song.ext`.  If I drop these folders into Navidrome, I'm going to have enormous amounts of albums with single tracks cluttering my database and making it hard to find things.

I'd love to use a dot-file to transform this into an album artist of "Various Artists", Year of "0000" and Album name of "My Awesome Playlist" so that it loads as an album, leaving the artist, track name, and track number as is. Doing this on import will ensure the original file remains intact with the album information (and will mean that it's kept in sync and won't re-download again).

I may in the end have to just overwrite the metadata of album artist (unnecessary), album name (moving it to TOAL), and year (moving it to originalyear) in the track metadata itself, but it seems like something that could reasonably be bypassed in a special case in the scanner logic.  **My bet is a lot of people have a folder full of singles that don't need to be shown as their own albums**

**Comment by deluan** [comment]
If we were to support more of these sidecar files, I think a better approach would be to support [Kodi-style .NFO files](https://kodi.wiki/view/NFO_files/Templates#album.nfo_Template). But don't quote me on this 🙃

**Reply by michaelkrieger** [reply]
_M-Tags ( see [HydrogenAud](https://wiki.hydrogenaud.io/index.php?title=Foobar2000:Components/m-TAGS_(foo_tags)) and [foobar2000 support](https://www.foobar2000.org/components/view/foo_tags) ) seems to have kind of never caught on as a standard with limited support. It was great as you could seed a file while modifying the metadata._

I’m fine with supporting m-tags or NFO files though if it solves this problem. Using those instead of reinventing the wheel makes sense for a sidecar file. A quick script could probably generate them for my use case, though I’d have to generate them for new files.

That said, may be better to integrate it in the regex-style ndignore files (as walked about elsewhere but maybe renamed to .navidrome) as a field where someone could designate an override.  This would be a more simple solution vs implementing a full new tag override system on a per file basis.

I do think a folder of singles is probably very common and this kind of solution allows for a clean library. 

**so I guess in summary, bonus points for that NFO file on a folder-basis :)**

appreciate your interest in the idea :)

**Reply by deluan** [reply]
It would not be a per file: The idea is that you could have a folder with a bunch of files, and have an `album.nfo` in it. This file would override the album info from the songs in the folder.

I think this is better than adding this (very proprietary) behaviour to Navidrome (.navidrome files, that is)

Anyway, currently none of these solutions are viable, as the current scanner will be replaced. Once the new one is fully integrated, we can think of these type of new features.

**Comment by michaelkrieger** [comment]
> The idea is that you could have a folder with a bunch of files, and have an `album.nfo` in it. This file would overrid the album info from the songs in the folder.

Sounds like a dream scenario.  Thank you for the interest in this.
Excited about this new scanner.

**Reply by certuna** [reply]
I've thought of a feature like this before - a server-side setting "DetectSingletons = true" which after the scan, cycles through each disc/album and uses the total tracks ("3/12") and if, say, more than half of the disc's tracks are missing, the songs are marked in the DB as "singleton = true", and then you can do things like make a section in the WebUI, exclude them from the album list, smart playlists, etc.

Main problem is, as mentioned, the OpenSubsonic API has no way to treat these tracks as different, so it would only work in the WebUI and in Smart Playlists. That's a discrepancy that's probably confusing for users.

**Comment by ChilledSlim** [comment]
This is awesome and much needed

**Comment by michaelkrieger** [comment]
@deluan @certuna any more thoughts on this one?  I have huge folders of singles that I’d love to squash the year and album tags on.

**Comment by deluan** [comment]
A solution could be adding support for sidecar files. I'll probably give some attention to [this PR](https://github.com/navidrome/navidrome/discussions/3181) after multi-library support. That could address your use-case.

**Reply by michaelkrieger** [reply]
Amazing!  Exciting to hear.
